### ISSUE TITLE
Cannot start services after updating from High Sierra to Big Sur

### ISSUE THREAD

**Issue by abhishek-pokhriyal** [laravel/valet#1036]
## Clear description of your problem
After updating from High Sierra to Big Sur, Valet stopped working.
![image](https://user-images.githubusercontent.com/11808845/107608909-4fb8b800-6c63-11eb-8c2b-e2c5c7ecd531.png)

## Expected behavior
It should as it working before.

## Diagnosis
<details>
<summary>sw_vers</summary>
<pre>ProductName:	macOS
ProductVersion:	11.2
BuildVersion:	20D64</pre>
</details>
<details>
<summary>valet --version</summary>
<pre>Laravel Valet 2.13.19</pre>
</details>
<details>
<summary>cat ~/.config/valet/config.json</summary>
<pre>{
    "tld": "test",
    "paths": [
        "/Users/abhishekpokhriyal/.config/valet/Sites"
    ]
}</pre>
</details>
<details>
<summary>cat ~/.composer/composer.json</summary>
<pre>{
    "require": {
        "laravel/valet": "^2.13",
        "phpro/grumphp": "^0.15.0",
        "friendsofphp/php-cs-fixer": "^2.15",
        "squizlabs/php_codesniffer": "*",
        "laravel/installer": "^4.0"
    }
}</pre>
</details>
<details>
<summary>composer global diagnose</summary>
<pre>Changed current directory to /Users/abhishekpokhriyal/.composer
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
require.squizlabs/php_codesniffer : unbound version constraints (*) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Composer version: 1.10.20
PHP version: 7.4.15
PHP binary path: /usr/local/Cellar/php@7.4/7.4.15/bin/php
OpenSSL version: OpenSSL 1.1.1i  8 Dec 2020</pre>
</details>
<details>
<summary>composer global outdated</summary>
<pre>Changed current directory to /Users/abhishekpokhriyal/.composer
composer/composer                  1.10.20 ~ 2.0.9  Composer helps you decla...
composer/semver                    1.7.2   ~ 3.2.4  Semver library that offe...
monolog/monolog                    1.26.0  ~ 2.2.0  Sends your logs to files...
php-cs-fixer/diff                  v1.3.1  ~ v2.0.2 sebastian/diff v2 backpo...
phpro/grumphp                      v0.15.2 ~ v1.3.1 A composer plugin that e...
symfony/config                     v4.4.19 ~ v5.2.3 Helps you find, load, co...
symfony/console                    v4.4.19 ~ v5.2.3 Eases the creation of be...
symfony/dependency-injection       v4.4.19 ~ v5.2.3 Allows you to standardiz...
symfony/event-dispatcher           v4.4.19 ~ v5.2.3 Provides tools that allo...
symfony/event-dispatcher-contracts v1.1.9  ~ v2.2.0 Generic abstractions rel...
symfony/filesystem                 v4.4.19 ~ v5.2.3 Provides basic utilities...
symfony/finder                     v4.4.19 ~ v5.2.3 Finds files and director...
symfony/options-resolver           v4.4.19 ~ v5.2.3 Provides an improved rep...
symfony/process                    v4.4.19 ~ v5.2.3 Executes commands in sub...
symfony/yaml                       v4.4.19 ~ v5.2.3 Loads and dumps YAML files</pre>
</details>
<details>
<summary>ls -al /etc/sudoers.d/</summary>
<pre>total 0
drwxr-xr-x   2 root  wheel    64 Jan  1  2020 .
drwxr-xr-x  87 root  wheel  2784 Feb 11 02:47 ..</pre>
</details>
<details>
<summary>brew config</summary>
<pre>HOMEBREW_VERSION: 3.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ebc7cc36a0e029d58fac8cebd1b5ff583b3fd95b
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d6c2977dba7ceb56c68141957c9ea5a16790c5e5
Core tap last commit: 21 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
xcrun: error: active developer path ("/Applications/Xcode.app/Contents/Developer") does not exist
Use `sudo xcode-select --switch path/to/Xcode.app` to specify the Xcode that you wish to use for command line developer tools, or use `xcode-select --install` to install the standalone command line developer tools.
See `man xcode-select` for more details.
Clang: N/A
Git: 2.30.1 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 1.8.0_231
macOS: 11.2-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A</pre>
</details>
<details>
<summary>brew services list</summary>
<pre>Name      Status  User Plist
dnsmasq   error   root /Library/LaunchDaemons/homebrew.mxcl.dnsmasq.plist
mysql@5.7 started root /Library/LaunchDaemons/homebrew.mxcl.mysql@5.7.plist
nginx     error   root /Library/LaunchDaemons/homebrew.mxcl.nginx.plist
php@7.4   error   root /Library/LaunchDaemons/homebrew.mxcl.php@7.4.plist
unbound   error   root /Library/LaunchDaemons/homebrew.mxcl.unbound.plist</pre>
</details>
<details>
<summary>brew list --formula --versions | grep -E "(php|nginx|dnsmasq|mariadb|mysql|mailhog|openssl)(@\d\..*)?\s"</summary>
<pre>dnsmasq 2.84
mysql@5.7 5.7.29
nginx 1.19.6
openssl@1.1 1.1.1i
php 7.4.8
php@7.4 7.4.15</pre>
</details>
<details>
<summary>brew outdated</summary>
<pre>dart-lang/dart/dart
gh
graphviz
ilmbase
imagemagick
jasper
jmeter
libomp
little-cms2
mysql@5.7
netpbm
openexr
openjpeg
pango
php
s3cmd
sass/sass/sass
watchman</pre>
</details>
<details>
<summary>brew tap</summary>
<pre>dart-lang/dart
homebrew/core
homebrew/services
sass/sass
shivammathur/php
wpengine/wpe-cli</pre>
</details>
<details>
<summary>php -v</summary>
<pre>PHP 7.4.15 (cli) (built: Feb  4 2021 17:22:59) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies</pre>
</details>
<details>
<summary>which -a php</summary>
<pre>/usr/local/opt/php@7.4/bin/php
/usr/local/bin/php
/usr/bin/php</pre>
</details>
<details>
<summary>php --ini</summary>
<pre>Configuration File (php.ini) Path: /usr/local/etc/php/7.4
Loaded Configuration File:         (none)
Scan for additional .ini files in: /usr/local/etc/php/7.4/conf.d
Additional .ini files parsed:      /usr/local/etc/php/7.4/conf.d/error_log.ini,
/usr/local/etc/php/7.4/conf.d/php-memory-limits.ini</pre>
</details>
<details>
<summary>nginx -v</summary>
<pre>nginx version: nginx/1.19.6</pre>
</details>
<details>
<summary>curl --version</summary>
<pre>curl 7.64.1 (x86_64-apple-darwin20.0) libcurl/7.64.1 (SecureTransport) LibreSSL/2.8.3 zlib/1.2.11 nghttp2/1.41.0
Release-Date: 2019-03-27
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS GSS-API HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets</pre>
</details>
<details>
<summary>php --ri curl</summary>
<pre>curl

cURL support => enabled
cURL Information => 7.75.0
Age => 8
Features
AsynchDNS => Yes
CharConv => No
Debug => No
GSS-Negotiate => No
IDN => Yes
IPv6 => Yes
krb4 => No
Largefile => Yes
libz => Yes
NTLM => Yes
NTLMWB => Yes
SPNEGO => Yes
SSL => Yes
SSPI => No
TLS-SRP => Yes
HTTP2 => Yes
GSSAPI => Yes
KERBEROS5 => Yes
UNIX_SOCKETS => Yes
PSL => No
HTTPS_PROXY => Yes
MULTI_SSL => Yes
BROTLI => Yes
Protocols => dict, file, ftp, ftps, gopher, gophers, http, https, imap, imaps, ldap, ldaps, mqtt, pop3, pop3s, rtmp, rtsp, scp, sftp, smb, smbs, smtp, smtps, telnet, tftp
Host => x86_64-apple-darwin20.2.0
SSL Version => (SecureTransport) OpenSSL/1.1.1i
ZLib Version => 1.2.11
libSSH Version => libssh2/1.9.0

Directive => Local Value => Master Value
curl.cainfo => no value => no value</pre>
</details>
<details>
<summary>~/.composer/vendor/laravel/valet/bin/ngrok version</summary>
<pre>ngrok version 2.3.35</pre>
</details>
<details>
<summary>ls -al ~/.ngrok2</summary>
<pre>ls: /Users/abhishekpokhriyal/.ngrok2: No such file or directory</pre>
</details>
<details>
<summary>brew info nginx</summary>
<pre>nginx: stable 1.19.6 (bottled), HEAD
HTTP(S) server and reverse proxy, and IMAP/POP3 proxy server
https://nginx.org/
/usr/local/Cellar/nginx/1.19.6 (25 files, 2.2MB) *
  Poured from bottle on 2021-02-11 at 11:04:52
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/nginx.rb
License: BSD-2-Clause
==> Dependencies
Required: openssl@1.1, pcre
==> Options
--HEAD
	Install HEAD version
==> Caveats
Docroot is: /usr/local/var/www

The default port has been set in /usr/local/etc/nginx/nginx.conf to 8080 so that
nginx can run without sudo.

nginx will load all files in /usr/local/etc/nginx/servers/.

To have launchd start nginx now and restart at login:
  brew services start nginx
Or, if you don't want/need a background service you can just run:
  nginx
==> Analytics
install: 35,886 (30 days), 124,917 (90 days), 453,913 (365 days)
install-on-request: 35,832 (30 days), 124,264 (90 days), 445,276 (365 days)
build-error: 0 (30 days)</pre>
</details>
<details>
<summary>brew info php</summary>
<pre>php: stable 8.0.2 (bottled), HEAD
General-purpose scripting language
https://www.php.net/
/usr/local/Cellar/php/7.4.8 (517 files, 76.4MB)
  Built from source
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/php.rb
License: PHP-3.01
==> Dependencies
Build: httpd, pkg-config
Required: apr, apr-util, argon2, aspell, autoconf, curl, freetds, gd, gettext, glib, gmp, icu4c, krb5, libffi, libpq, libsodium, libzip, oniguruma, openldap, openssl@1.1, pcre2, sqlite, tidy-html5, unixodbc
==> Options
--HEAD
	Install HEAD version
==> Caveats
To enable PHP in Apache add the following to httpd.conf and restart Apache:
    LoadModule php_module /usr/local/opt/php/lib/httpd/modules/libphp.so

    <FilesMatch \.php$>
        SetHandler application/x-httpd-php
    </FilesMatch>

Finally, check DirectoryIndex includes index.php
    DirectoryIndex index.php index.html

The php.ini and php-fpm.ini file can be found in:
    /usr/local/etc/php/8.0/

To have launchd start php now and restart at login:
  brew services start php
Or, if you don't want/need a background service you can just run:
  php-fpm
==> Analytics
install: 56,123 (30 days), 163,708 (90 days), 621,569 (365 days)
install-on-request: 55,009 (30 days), 160,388 (90 days), 599,670 (365 days)
build-error: 0 (30 days)</pre>
</details>
<details>
<summary>brew info openssl</summary>
<pre>openssl@1.1: stable 1.1.1i (bottled) [keg-only]
Cryptography and SSL/TLS Toolkit
https://openssl.org/
/usr/local/Cellar/openssl@1.1/1.1.1i (8,067 files, 18.5MB)
  Poured from bottle on 2021-02-11 at 02:11:23
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/openssl@1.1.rb
License: OpenSSL
==> Caveats
A CA file has been bootstrapped using certificates from the system
keychain. To add additional certificates, place .pem files in
  /usr/local/etc/openssl@1.1/certs

and run
  /usr/local/opt/openssl@1.1/bin/c_rehash

openssl@1.1 is keg-only, which means it was not symlinked into /usr/local,
because macOS provides LibreSSL.

If you need to have openssl@1.1 first in your PATH, run:
  echo 'export PATH="/usr/local/opt/openssl@1.1/bin:$PATH"' >> ~/.zshrc

For compilers to find openssl@1.1 you may need to set:
  export LDFLAGS="-L/usr/local/opt/openssl@1.1/lib"
  export CPPFLAGS="-I/usr/local/opt/openssl@1.1/include"

For pkg-config to find openssl@1.1 you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/openssl@1.1/lib/pkgconfig"

==> Analytics
install: 704,506 (30 days), 2,250,333 (90 days), 8,228,943 (365 days)
install-on-request: 108,669 (30 days), 355,635 (90 days), 1,191,286 (365 days)
build-error: 0 (30 days)</pre>
</details>
<details>
<summary>openssl version -a</summary>
<pre>LibreSSL 2.8.3
built on: date not available
platform: information not available
options:  bn(64,64) rc4(16x,int) des(idx,cisc,16,int) blowfish(idx) 
compiler: information not available
OPENSSLDIR: "/private/etc/ssl"</pre>
</details>
<details>
<summary>openssl ciphers</summary>
<pre>ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-CHACHA20-POLY1305:GOST2012256-GOST89-GOST89:DHE-RSA-CAMELLIA256-SHA256:DHE-RSA-CAMELLIA256-SHA:GOST2001-GOST89-GOST89:AES256-GCM-SHA384:AES256-SHA256:AES256-SHA:CAMELLIA256-SHA256:CAMELLIA256-SHA:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-RSA-CAMELLIA128-SHA256:DHE-RSA-CAMELLIA128-SHA:AES128-GCM-SHA256:AES128-SHA256:AES128-SHA:CAMELLIA128-SHA256:CAMELLIA128-SHA:ECDHE-RSA-RC4-SHA:ECDHE-ECDSA-RC4-SHA:RC4-SHA:RC4-MD5:ECDHE-RSA-DES-CBC3-SHA:ECDHE-ECDSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:DES-CBC3-SHA</pre>
</details>
<details>
<summary>sudo nginx -t</summary>
<pre>nginx: the configuration file /usr/local/etc/nginx/nginx.conf syntax is ok
nginx: configuration file /usr/local/etc/nginx/nginx.conf test is successful</pre>
</details>
<details>
<summary>which -a php-fpm</summary>
<pre>/usr/local/sbin/php-fpm
/usr/local/opt/php@7.4/sbin/php-fpm
/usr/sbin/php-fpm</pre>
</details>
<details>
<summary>/usr/local/opt/php/sbin/php-fpm -v</summary>
<pre>PHP 7.4.8 (fpm-fcgi) (built: Jul  9 2020 23:46:44)
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies</pre>
</details>
<details>
<summary>sudo /usr/local/opt/php/sbin/php-fpm -y /usr/local/etc/php/7.4/php-fpm.conf --test</summary>
<pre>[11-Feb-2021 12:15:53] ERROR: failed to open configuration file '/usr/local/etc/php/7.4/php-fpm.conf': No such file or directory (2)
[11-Feb-2021 12:15:53] ERROR: failed to load configuration file '/usr/local/etc/php/7.4/php-fpm.conf'
[11-Feb-2021 12:15:53] ERROR: FPM initialization failed</pre>
</details>
<details>
<summary>ls -al ~/Library/LaunchAgents | grep homebrew</summary>
<pre>-rw-r--r--   1 abhishekpokhriyal  staff   657 Feb 11 12:13 homebrew.mxcl.dnsmasq.plist
-rw-r--r--   1 abhishekpokhriyal  staff   551 Feb 11 12:13 homebrew.mxcl.mysql@5.7.plist
-rw-r--r--   1 abhishekpokhriyal  staff   571 Feb 11 12:13 homebrew.mxcl.nginx.plist
-rw-r--r--   1 abhishekpokhriyal  staff   636 May 25  2020 homebrew.mxcl.php@7.2.plist
-rw-r--r--   1 abhishekpokhriyal  staff   636 Feb 11 02:07 homebrew.mxcl.php@7.3.plist
-rw-r--r--   1 abhishekpokhriyal  staff   636 Feb 11 12:13 homebrew.mxcl.php@7.4.plist
-rw-r--r--   1 abhishekpokhriyal  staff   688 Feb 11 12:13 homebrew.mxcl.unbound.plist</pre>
</details>
<details>
<summary>ls -al /Library/LaunchAgents | grep homebrew</summary>
<pre></pre>
</details>
<details>
<summary>ls -al /Library/LaunchDaemons | grep homebrew</summary>
<pre>-rw-r--r--   1 root  admin   657 Feb 11 12:13 homebrew.mxcl.dnsmasq.plist
-rw-r--r--   1 root  admin   551 Feb 11 12:13 homebrew.mxcl.mysql@5.7.plist
-rw-r--r--   1 root  admin   571 Feb 11 12:13 homebrew.mxcl.nginx.plist
-rw-r--r--   1 root  admin   636 May 25  2020 homebrew.mxcl.php@7.2.plist
-rw-r--r--   1 root  admin   636 Feb 11 12:13 homebrew.mxcl.php@7.4.plist
-rw-r--r--   1 root  admin   688 Feb 11 12:13 homebrew.mxcl.unbound.plist</pre>
</details>
<details>
<summary>ls -aln /etc/resolv.conf</summary>
<pre>lrwxr-xr-x  1 0  0  22 Jan  1  2020 /etc/resolv.conf -> ../var/run/resolv.conf</pre>
</details>
<details>
<summary>cat /etc/resolv.conf</summary>
<pre>#
# macOS Notice
#
# This file is not consulted for DNS hostname resolution, address
# resolution, or the DNS query routing mechanism used by most
# processes on this system.
#
# To view the DNS configuration used by this system, use:
#   scutil --dns
#
# SEE ALSO
#   dns-sd(1), scutil(8)
#
# This file is automatically generated.
#
nameserver 8.8.8.8
nameserver 192.168.0.10</pre>
</details>

**Comment by jools-r** [comment]
Your brew config shows that the Xcode command line tools are not installed, or at least that the path is wrong:

```
xcrun: error: active developer path ("/Applications/Xcode.app/Contents/Developer") does not exist
Use `sudo xcode-select --switch path/to/Xcode.app` to specify the Xcode that you wish to use for command line developer tools, or use `xcode-select --install` to install the standalone command line developer tools.
See `man xcode-select` for more details.
```

If you know where your Xcode installation is, you may want to switch as advised in the comment. Personally, I've had to reinstall the Xcode Command line tools for every Big Sur update so far: from 11.0->11.1, 11.1->11.2 and even 11.2->11.2.1 as follows: 

```
xcode-select --install
```

That may suffice for you, but I found I additionally had to reinstall git afterwards as described here: https://github.com/Homebrew/discussions/discussions/673#discussioncomment-345111.

**Comment by abhishek-pokhriyal** [comment]
Did that. Now running `xcode-select --install` gives:
![image](https://user-images.githubusercontent.com/11808845/107622737-41c46080-6c7e-11eb-98a9-3d4370be785c.png)

**Comment by abhishek-pokhriyal** [comment]
Tried so many things but could get it to work:

![image](https://user-images.githubusercontent.com/11808845/107642121-73e3bb80-6c9a-11eb-98e2-b2ee9819c596.png)

**Comment by drbyte** [comment]
1. Your global composer appears to be still using 1.x
I suggest:
`composer global update --with-all-dependencies`
and
`composer self-update`

2. You've got all your Homebrew services set up to run both as root and non-root users. This has them competing against each other.
Based on the list you provided, you need to run the following commands to stop the competing services:
```
brew services stop dnsmasq
brew services stop nginx
brew services stop php@7.2
brew services stop php@7.3
brew services stop php@7.4
sudo brew services stop mysql@5.7
sudo brew services stop php@7.2
sudo brew services stop php@7.4
sudo brew services stop unbound
```
Then, run the following to rebuild some things that appear to be incomplete:
```
valet stop
valet install
valet use php@7.4 --force
```

3. You have a lot of Homebrew apps that are outdated. You should upgrade those regularly.
I recommend on a monthly or weekly basis to run `brew upgrade`.

In fact, **it is good practice to run the following weekly:**
```
composer self-update
composer global update --with-all-dependencies
brew upgrade
```

**Comment by abhishek-pokhriyal** [comment]
First up, @drbyte Thank you for the response and for pointing those important issues out. Really appreciate it.

I started following your step and found `composer self-update` did not work (failed with `Command "self-update" is not defined.`)

So I just updated the version of the composer from their official documentation (and Laracasts 😅).

All the other steps proceeded fine but I was still getting 502 when I was visiting the link in the browser. So, I checked the nginx logs and found this

> 2021/02/12 00:48:55 [crit] 5452#0: *27 connect() to unix:/Users/abhishekpokhriyal/.config/valet/valet.sock failed (2: No such file or directory) while connecting to upstream, client: 127.0.0.1, server: my-site.test, request: "GET / HTTP/2.0", upstream: "fastcgi://unix:/Users/abhishekpokhriyal/.config/valet/valet.sock:", host: "my-site.test"

And then suddenly I recalled, I've seen this error before. I searched the list of issues and Voilà! I found this: https://github.com/laravel/valet/issues/865#issuecomment-661907758 – my own answer!

Was definitely a tough day at work today but I'm happy that I made it and will sleep well. Thank you for all the effort you put in responding to queries, DrByte.
